### PR TITLE
Y.F.C Implement Asynchronous Fence manager and Rework Query async downloads

### PR DESCRIPTION
**Pull request by FernandoS27**
This pull requests aims to do 3 things:

1st Allow verification of fencing and writing of async downloads in a separate thread.
2nd Restructure the way accuracy is handled. Normal accuracy will now skip host-guest fence syncing and won't download on host conditional rendering.
3rd Rework Query Cache async downloads to be more consistent.

as a result when paired with the buffer cache revamp and Faster DMA:

- Performance greatly improves in many games with Normal accuracy.
- Splatoon 2's ink now works perfectly in AMD. (Splatoon 2 requires High accuracy).

### CODE REVIEW

**Comment by gidoly** [comment]
breaks nier automata


https://user-images.githubusercontent.com/66776795/233872436-ca3dc5a9-5e5a-4ce8-b552-4f9efc1ab2ff.mp4

**Comment by gidoly** [comment]
https://user-images.githubusercontent.com/66776795/233874954-411775a7-3bbd-42ba-8e97-ee76608d3d0c.mp4

 cloud on xc flickers would be nice to get fix as game benifit lot from normal accuracy

**Comment by Kerobotto** [comment]
image
Water seems to be affected in BOTW. Tried resetting my shaders but it didn't change anything.

**Comment by FernandoS27** [comment]
> breaks nier automata
> 
>  yuzu_red_pill_xVhDmUBvgK.mp4

fixed.

File: src/video_core/fence_manager.h [hunk 1]
```diff
@@ -46,17 +64,33 @@ class FenceManager {
...
+            guard.lock();
+        }
+        if (delay_fence) {
+            uncommitted_operations.emplace_back(std::move(func));
+        }
+        pending_operations.emplace_back(std::move(uncommitted_operations));
         QueueFence(new_fence);
+        if (!delay_fence) {
+            func();
+        }
```
Comment (Kelebek1) [inline src/video_core/fence_manager.h:85]: This cuts some games' FPS in half. Setting delay fence to perma true fixes it, as does removing this branch and always using the uncommitted_operations one above. The functions called here don't seem to be very costly (SyncpointManager::IncrementHost and some u32 memory writes) so I'm not sure why.

**Comment by Lycoris2013** [comment]
> yuzu_W6saaaskiA.mp4 
>  explorer_ztuePgReRY.mp4 
> cloud on xc flickers would be nice to get fix as game benifit lot from normal accuracy which not happened on original accuracy normal. issue only happens when bcrr pr and this pr combined seperately it doesn't cause issue not happens on blue pill internal build

This was resolved with these commits. I will leave a comment here as well.
9a7c172f76007324d72d107acc01c64e41d53820
2f158765240623d5aea916cafca71b86be2d6fd4

**Review by bunnei — APPROVED** [review]

**Comment by Lycoris2013** [comment]
EA3549 and the blinking occurred. The following build does not cause the problem.
https://github.com/yuzu-emu/yuzu/actions/runs/4835076952
I think #10088 has other conflict.

The following settings will reduce the blinking, even in normal mode.
nVidia Driver Settings: Vulkan/OpenGL present method->Prefer layered on DXGI Swapchain

**Pull request merged**